SimTKcommon/src/Pathname.cpp

```diff
@@ -238,6 +240,161 @@ void Pathname::deconstructPathname( const string&   name,
     }
 }
 
+void Pathname::deconstructPathnameUsingSpecifiedWorkingDirectory(const std::string& swd,
+                                                                 const std::string& path,
+                                                                 std::string& directory,
+                                                                 std::string& fileName,
+                                                                 std::string& extension)
+{
+    directory.erase(); fileName.erase(); extension.erase();
+    string pathdrive, swddrive, finaldrive;
+    String processed;
+
+    // Remove all the white space and make all the slashes be forward ones.
+    // (For Windows they'll be changed to backslashes later.)
+    String pathcleaned = String::trimWhiteSpace(path).replaceAllChar('\\', '/');
+    if (pathcleaned.empty())
+        return; // path consisted only of white space
```

Comment:
Hm. I keep going back and forth with this. Given how I envisioned the use cases, I didn't think an empty path would be okay. I imagine any use like this should have just used getCurrentWorkingDirectory() anyway? For now I'm going to keep this as is and then add some more comments.